src/LowLevel/SecurityToken.ts

```diff
@@ -205,6 +237,26 @@ export class SecurityToken extends Contract<SecurityTokenContract> {
     return addresses.map(address => new UsdTieredSto({ address, context }));
   }
 
+  /**
+   * Given STO module type and address, this function will return a generic, LowLevel Sto object.
+   *
+   * @TODO remon-nashid: shouldn't module address be enough to fetch a module?
+   */
+  public getSTOModule = async ({ address, stoType }: GetSTOModuleArgs): Promise<Sto | null> => {
+    const { context } = this;
+    const { methods } = this.contract;
+
+    const { 3: isArchived } = await methods.getModule(address).call();
```

Comment:
You should check if the module actually exists and is attached (checking if the `module` property is different from `ZERO_ADDRESS`). Otherwise, return null.

You should also check if it is of the corresponding type (converting the `name` property from hex to ascii and, depending on the type, checking if it is `CappedSTO` or `USDTieredSTO`). Otherwise, throw an error.

Reply:
I was sticking to the pattern of validating user input, and throwing if needed, in Procedures. 

As for module type, I'll rely on the in returned `ModuleData` struct instead of user input.